sim/scenarios/droplist/droplist-error-model.cc

```diff
@@ -0,0 +1,35 @@
+#include <map>
+#include "droplist-error-model.h"
+
+using namespace std;
+
+NS_OBJECT_ENSURE_REGISTERED(DroplistErrorModel);
+ 
+TypeId DroplistErrorModel::GetTypeId(void) {
+  static TypeId tid = TypeId("DroplistErrorModel")
+    .SetParent<ErrorModel>()
+    .AddConstructor<DroplistErrorModel>()
+    ;
+  return tid;
+}
+ 
+DroplistErrorModel::DroplistErrorModel() { }
+
+void DroplistErrorModel::DoReset(void) { }
+ 
+bool DroplistErrorModel::DoCorrupt(Ptr<Packet> p) {
+  static int packet_num = 0;
+  return drops.empty() ? 0 : drops.find(++packet_num) != drops.end();
+}
+
+void DroplistErrorModel::SetDrops(string drops_in) {
+  char *cstr = new char[drops_in.length()+1];
+  strcpy(cstr, drops_in.c_str());
+  char *p = strtok(cstr,",");
+  while (p) {
+    drops.insert(make_pair(stoi(p), true));
+    p = strtok(NULL,",");
+  }
+  delete[] cstr;
```

Comment:
This looks like C. Maybe this is more idiomatic C++?
```cpp
while(true) {
    static std::size_t last_found = 0;
    std::size_t found = str.find(",", last_found);
    drops.insert(stoi(str.substr(last_found, found-last_found)));
    if(found == std::string::npos) break;
    last_found = found + 1;
  }
```
But it doesn't look nicer. So I'm fine with leaving it as it is.

Reply:
Either works for me, but yes you are right that this is basically C. I went with what naturally comes to me. I'll leave it as is.